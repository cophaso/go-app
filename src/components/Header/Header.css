.Header {
  height: 40px;
  display: flex;
  align-items: normal;
  justify-content: space-between;
}

.Header h1 {
  margin: 0;
  padding-left: 20px;
}

.Header a {
  color: #444;
  text-decoration: none;
  padding-right: 10px;
}

.Header__tagline--wide {
  opacity: 0.35;
  margin-left: 10px;
}

.Header__tagline--narrow {
  opacity: 0.35;
  display: block;
  text-align: center;
  display: none;
  margin-top: 10px;
}

.Header__not-logged-in {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: flex-end;
  padding-right: 10px;
}

.Header__not-logged-in a:last-child {
  background: rgb(78, 159, 230);
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
}

.Header__logged-in a {
  border: 1px solid rgba(0, 0, 0, 0.25);
  padding: 5px 15px;
  border-radius: 15px;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  padding-top: 10px;
  padding-bottom: 50px;
  background-color: #aaa;
}

@media screen and (min-width: 450px) {
  .Header {
    align-items: baseline;
  }

  .Header h1 {
    font-size: 22pt;
  }
}

@media screen and (max-width: 490px) {
  .Header__tagline--wide {
    display: none;
  }

  .Header__tagline--narrow {
    display: inherit;
  }
}
