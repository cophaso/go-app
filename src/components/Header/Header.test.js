import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe(`Header component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a Header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the Header given props', () => {
    const wrapper = shallow(<Header {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
