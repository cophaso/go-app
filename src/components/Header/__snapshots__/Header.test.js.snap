// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<Fragment>
  <nav
    className="Header"
  >
    <h1>
      <Link
        to="/"
      >
         
        Go!
      </Link>
    </h1>
    <div
      className="Header__not-logged-in"
    >
      <Link
        to="/login"
      >
        Log in
      </Link>
      <Link
        to="/signup"
      >
        Sign Up
      </Link>
    </div>
  </nav>
</Fragment>
`;

exports[`Header component renders the Header given props 1`] = `
<Fragment>
  <nav
    className="Header"
  >
    <h1>
      <Link
        to="/"
      >
         
        Go!
      </Link>
    </h1>
    <div
      className="Header__not-logged-in"
    >
      <Link
        to="/login"
      >
        Log in
      </Link>
      <Link
        to="/signup"
      >
        Sign Up
      </Link>
    </div>
  </nav>
</Fragment>
`;
